lib/rucio/daemons/conveyor/receiver.py

```diff
@@ -120,130 +113,70 @@ def _perform_request_update(
-                    METRICS.counter('reconnect.{host}').labels(host=conn.transport._Transport__host_and_ports[0][0].split('.')[0]).inc()
-
-                    conn.set_listener(
-                        'rucio-messaging-fts3',
-                        Receiver(
-                            broker=conn.transport._Transport__host_and_ports[0],
-                            id_=id_,
-                            total_threads=total_threads,
-                            transfer_stats_manager=transfer_stats_manager,
-                            all_vos=all_vos
-                        ))
-                    if not use_ssl:
-                        conn.connect(username, password, wait=True)
-                    else:
-                        conn.connect(wait=True)
-                    conn.subscribe(destination=config_get('messaging-fts3', 'destination'),
-                                   id='rucio-messaging-fts3',
-                                   ack='auto')
+            conn_mgr.subscribe(id_='rucio-messaging-fts3', ack='auto')
+
             time.sleep(1)
 
-        for conn in conns:
-            try:
-                conn.disconnect()
-            except Exception:
-                pass
+        conn_mgr.disconnect()
 
 
-def stop(signum: Optional[int] = None, frame: Optional["FrameType"] = None) -> None:
+def stop(signum: int | None = None, frame: FrameType | None = None) -> None:
     """
     Graceful exit.
     """
 
     GRACEFUL_STOP.set()
 
 
-def run(
-        once: bool = False,
-        total_threads: int = 1
-) -> None:
+def run(once: bool = False, total_threads: int = 1) -> None:
     """
     Starts up the receiver thread
     """
     setup_logging(process_name=DAEMON_NAME)
+    logger = formatted_logger(logging.log, DAEMON_NAME + ' %s')
 
     if rucio.db.sqla.util.is_old_db():
         raise exception.DatabaseException('Database was not updated, daemon won\'t start')
 
-    logging.info('starting receiver thread')
-    threads = [threading.Thread(target=receiver, kwargs={'id_': i,
-                                                         'total_threads': total_threads}) for i in range(0, total_threads)]
+    logger(logging.INFO, 'starting receiver thread')
+    threads = []
+    for i in range(total_threads):
+        rec_thread = threading.Thread(target=receiver,
+                                      kwargs={'id_': i, 'logger': logger, 'total_threads': total_threads})
+        rec_thread.start()
+        threads.append(rec_thread)
 
-    [thread.start() for thread in threads]
-
-    logging.info('waiting for interrupts')
+    logger(logging.INFO, 'waiting for interrupts')
 
     # Interruptible joins require a timeout.
-    while threads:
-        threads = [thread.join(timeout=3.14) for thread in threads if thread and thread.is_alive()]
+    # while alive_threads := [thread for thread in threads if thread.is_alive()]:
+    #     for thread in alive_threads:
+    #         thread.join(timeout=3.14)
```

Comment:
Commented out code should be removed or a comment should be added with the reason for why it must be kept in production